### PR TITLE
feat(stdlib): Add Array.reverse() function

### PR DESCRIPTION
**Pull request by technosophos**
This adds an `Array.reverse()` function that takes a source array and returns a new array with the elements reversed. It also adds a test in the compiler tests.

This helps bring `Array` and `List` closer to parity.

I am not sure where to put the documentation for this function.

Signed-off-by: Matt Butcher <matt.butcher@microsoft.com>

### CODE REVIEW

**Review by ospencer — APPROVED** [review]
Looks great! There's maybe a question of if this should happen in place, but I think we'll probably want both anyway. I think @phated might have ideas about this (since no one seems to like my idea of following the Ruby tradition of bang names, i.e. `Array.reverse!()` for functions that mutate the argument 😂).

Also, no need to worry about the test failure—that's fixed in #700. You may just need to rebase to bring in that change.

File: compiler/test/stdlib/array.test.gr [hunk 1]
```diff
@@ -94,6 +94,11 @@ Array.forEachi(addToSum, arr1)
 
 assert sum == 8
 
+// Array.reverse
+
+assert Array.reverse([> ]) == [> ] // Empty
+assert Array.reverse([> 1, 2, 3, 4]) == [> 4, 3, 2, 1]
```
Comment (peblair) [inline compiler/test/stdlib/array.test.gr:100]: Based on @ospencer's review, I might suggest an explicit test for this not being in-place; just so we can make sure that behavior doesn't change in the future
```suggestion
let testArr = [> 1, 2, 3, 4]
assert Array.reverse(testArr) == [> 4, 3, 2, 1] // array is reversed properly
assert testArr == [> 1, 2, 3, 4] // original array is not modified
```

**Review by peblair — CHANGES_REQUESTED** [review]
Thanks for adding this! The function itself looks great, but I think we should go ahead and check that we don't modify the original array in the test suite. Outside of that, this looks good to go!

File: compiler/test/stdlib/array.test.gr (hunk 1 above)
Comment (technosophos) [reply compiler/test/stdlib/array.test.gr:100]: Good idea.

**Comment by technosophos** [comment]
I was never a huge Ruby fan, but I did like the concise nature of the `!` for modify-in-place. That said, I'm cool with whatever convention you adopt.

I can also verify that after rebasing with #700, I no longer get the failed test.

**Review by phated — APPROVED** [review]
Hooray! 🎉

**Pull request merged**